llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/IR/LinalgExtOps.cpp

```diff
@@ -1447,6 +1447,118 @@ LogicalResult TopkOp::getResultTilePosition(
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// PackOp
+//===----------------------------------------------------------------------===//
+
+LogicalResult PackOp::verify() {
+  // input rank = r
+  // output rank = 2 * r (input rank + tile rank).
+  // inner tiles = r
+  return success();
+}
+
+SmallVector<OpFoldResult> PackOp::getMixedInnerTiles() {
+  SmallVector<OpFoldResult> mixedInnerTiles;
+  mixedInnerTiles.reserve(getInputRank());
+  unsigned dynamicValIndex = 0;
+  for (Attribute attr : getStaticInnerTiles()) {
+    auto tileAttr = attr.cast<IntegerAttr>();
+    if (!ShapedType::isDynamic(tileAttr.getInt()))
+      mixedInnerTiles.push_back(tileAttr);
+    else
+      mixedInnerTiles.push_back(getInnerTiles()[dynamicValIndex++]);
+  }
+  return mixedInnerTiles;
+}
+
+SmallVector<StringRef> PackOp::getLoopIteratorTypes() {
```

Comment:
This looks more like a linalg op than a tensor op.
Despite it living in linalg_ext for now, I am thinking that this would live close to `tensor.pad`.

Reply:
ah I missed that it was part of the tilinginterface, please move to a comment-delimited region that says tiling interface.